CHANGES.md

```diff
@@ -0,0 +1,86 @@
+# Revision history for sDDF
+
+## Release 0.5.0
+
+### General
+
+* Move to Microkit 1.4.0.
+    * Previously a development version of Microkit was used, now we use
+      an official release for everything.
+* Add QEMU support to the timer and echo server examples.
+* Remove dependency on a full libc, all the provided components use our own
+  utility functions now.
+* Moved all components and examples to use a 'Makefile snippets' system
+  to make it easier to put together sDDF components in a system.
+* Introduced snippets for the Zig build system, as an alternative to Make.
+    * You can find instructions for using the Zig build system in each example's
+      README.
+    * No components are solely built with Zig, the primary build system is
+      still Make.
+
+### Block
+
+* Move the capacity of the shared queue out of shared memory
+  and into the queue handles. This prevents malicious clients
+  from messing with the queue size and potentially causing the
+  virtualiser or other trusted components to crash/misbehave.
+* Introduce a config header to enable the system designer to have a variable
+  amount of clients, previously it was all hard-coded.
+    * It should be noted that we are working on tooling to automate this process
+      and these configs are likely to significantly change by the next release.
+* Change block virtualiser to use offsets instead of raw addresses and do the
+  offset to DMA physical addresses in the virtualiser.
+    * This makes the block system consistent with the networking system in how it
+      handles client-to-driver address translation.
+* Use 'capacity' instead of size in the shared queue library to avoid confusion.
+
+### I2C
+
+* Use 'capacity' instead of size in the shared queue library to avoid confusion.
+
+### Network
+
+* Add a virtIO network driver.
+    * This allows us to have networking on QEMU making it easier to simulate
+      non-trivial systems.
+* Fix caching operations in RX virtualiser, previously this would lead to
+  issues with TCP traffic.
+* Various bug fixes in the Amlogic ethernet driver.
+* Fixes and changes for improved TCP performance.
+* Add support for broadcast packets in the TX virtualiser, previously
```

Comment:
RX* not TX

Reply:
Also, I wouldn't day incorrectly. This was an intentional choice, but we didn't realise that those could result in difficulty with dhcp traffic. I would more say that we changed our default implementation of broadcast traffic handling from being sent to a special broadcast component (arp) to instead be forwarded to all clients, allowing us to correctly handle certain networking protocols making use of the broadcast MAC address.

Reply:
Sure thing, fixed it now.